src/wallet/rpcwallet.cpp

```diff
@@ -2399,6 +2400,11 @@ static UniValue getbalances(const JSONRPCRequest& request)
     const auto bal = wallet.GetBalance();
     UniValue balances{UniValue::VOBJ};
     {
+        Optional<int> height = locked_chain->getHeight();
+        if (height) {
+            uint256 lastblock = locked_chain->getBlockHash(height.get());
```

Comment:
this is incorrect, I believe. The chain can proceed to the next block after the call to BlockUntilSyncedToCurrentChain, but the wallet may not catch up until the lock is taken in the next line.

This should be wallet.m_best_block_hash

Reply:
Thanks!

Have changed the code by introducing a new wallet function **GetLastBlockHash()** that returns `m_last_block_processed `.